src/python/director/viewerclient.py

```diff
@@ -0,0 +1,240 @@
+def serialize_transform(tform):
+    return {
+        "translation": list(transformations.translation_from_matrix(tform)),
+        "quaternion": list(transformations.quaternion_from_matrix(tform))
+    }
+
+
+class GeometryData:
+    __slots__ = ["geometry", "color", "transform"]
+    def __init__(self, geometry, color=[1, 0, 0, 0.5], transform=np.eye(4)):
+        self.geometry = geometry
+        self.color = color
+        self.transform = transform
+
+    def serialize(self):
+        params = self.geometry.serialize()
+        params["color"] = self.color
+        params["transform"] = serialize_transform(self.transform)
+        return params
+
+
+class BaseGeometry:
+    def serialize(self):
+        raise NotImplementedError()
+
+
+class Box(BaseGeometry, namedtuple("Box", ["lengths"])):
+    def serialize(self):
+        return {
+            "type": "box",
+            "lengths": list(self.lengths)
+        }
+
+
+class Sphere(BaseGeometry, namedtuple("Sphere", ["radius"])):
+    def serialize(self):
+        return {
+            "type": "sphere",
+            "radius": self.radius
+        }
+
+
+class Ellipsoid(BaseGeometry, namedtuple("Ellipsoid", ["radii"])):
+    def serialize(self):
+        return {
+            "type": "ellipsoid",
+            "radii": list(self.radii)
+        }
+
+
+class Cylinder(BaseGeometry, namedtuple("Cylinder", ["length", "radius"])):
+    def serialize(self):
+        return {
+            "type": "cylinder",
+            "length": self.length,
+            "radius": self.radius
+        }
+
+
+class Triad(BaseGeometry, namedtuple("Triad", [])):
+    def serialize(self):
+        return {
+            "type": "triad"
+        }
+
+
+class LazyTree:
+    __slots__ = ["geometries", "transform", "children"]
+    def __init__(self, geometries=[], transform=np.eye(4)):
+        self.geometries = geometries
+        self.transform = transform
+        self.children = defaultdict(lambda: LazyTree())
+
+    def __getitem__(self, item):
+        return self.children[item]
+
+    def getdescendant(self, path):
+        t = self
+        for p in path:
+            t = t[p]
+        return t
+
+
+class CommandQueue:
+    def __init__(self):
+        self.draw = set()
+        self.load = set()
+        self.delete = set()
+
+    def isempty(self):
+        return not (self.draw or self.load or self.delete)
+
+
+class Visualizer:
+    __slots__ = ["core", "path"]
+    def __init__(self, core=None, path=None, lcm=LCM()):
+        if core is None:
+            core = CoreVisualizer(lcm)
+        if path is None:
+            path = tuple()
+        self.core = core
+        self.path = path
+
+    def load(self, geomdata):
+        self.core.load(self.path, geomdata)
+        return self
+
+    def draw(self, tform):
+        self.core.draw(self.path, tform)
+
+    def delete(self):
+        self.core.delete(self.path)
+
+    def __getitem__(self, path):
+        return Visualizer(self.core, self.path + (path,))
```

Comment:
Yeah, that's probably better (it currently has no effect, since that `lcm` arg is only used if `core` is not passed, but "explicit is better than implicit")